lib/framework/output.ts

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2020 Balena
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import { getCliUx, getChalk } from '../utils/lazy';
+
+export interface DataOutputOptions {
```

Comment:
> Also: this interface may a good opportunity to introduce another framework-level flag:

I'm not a fan of expanding the scope of PRs like that.  We should do that in a separate PR after this gets merged.

Reply:
> Would the word Flags (instead of Options) make it clearer that these are the --fields, --json, --verbose 

I chose options here on purpose. Reason being is that I think we should standardize on `flags` for direct cli input, and `options` for the general more abstract use of options not directly at the UI level.

The `DataOutputOptions` interface is arguably used more generally (e.g. in the output module) - it maps to cli flags, but doesn't necessarily need to be that, and doesn't have any UI flag concerns.   Whereas the CLI flags themselves are of type `flags.Input<DataOutputOptions>`, and the set of flags is called `dataOutputFlags` (it does contain UI concerns).  (Not sure if I've explained my reasoning very clearly there).